docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,617 @@
+
+### MissingMemberHandling
+
+`Newtonsoft.Json` can be configured to throw exceptions during deserialization if the JSON includes properties that are missing in the target type. <xref:System.Text.Json> ignores extra properties in the JSON, except when you use the [[JsonExtensionData] attribute](system-text-json-how-to.md#handle-overflow-json). There's no workaround for the missing member feature.
+
+### TraceWriter
+
+`Newtonsoft.Json` lets you debug by using a `TraceWriter` to view logs that are generated by serialization or deserialization. <xref:System.Text.Json> doesn't do logging.
+
+## JsonDocument and JsonElement compared to JToken (like JObject, JArray)
+
+<xref:System.Text.Json.JsonDocument?displayProperty=fullName> provides the ability to parse and build a **read-only** Document Object Model (DOM) from existing JSON payloads. The DOM provides random access to data in a JSON payload. The JSON elements that compose the payload can be accessed via the <xref:System.Text.Json.JsonElement> type. The `JsonElement` type provides APIs to convert JSON text to common .NET types. `JsonDocument` exposes a <xref:System.Text.Json.JsonDocument.RootElement> property.
+
+### JsonDocument is IDisposable
+
+`JsonDocument` builds an in-memory view of the data into a pooled buffer. Therefore, unlike `JObject` or `JArray` from `Newtonsoft.Json`, the `JsonDocument` type implements `IDisposable` and needs to be used inside a using block. 
+
+Only return a `JsonDocument` from your API if you want to transfer lifetime ownership and dispose responsibility to the caller. In most scenarios, that isn't necessary. If the caller needs to work with the entire JSON document, return the <xref:System.Text.Json.JsonElement.Clone%2A> of the <xref:System.Text.Json.JsonDocument.RootElement%2A>, which is a <xref:System.Text.Json.JsonElement>. If the caller needs to work with a particular element within the JSON document, return the <xref:System.Text.Json.JsonElement.Clone%2A> of that <xref:System.Text.Json.JsonElement>. If you return the `RootElement` or a sub-element directly without making a `Clone`, the caller won't be able to access the returned `JsonElement` after the `JsonDocument` that owns it is disposed.
+
+Here's an example that requires you to make a `Clone`:
+
+```csharp
+public JsonElement LookAndLoad(JsonElement source)
+{
+    string json = File.ReadAllText(source.GetProperty("fileName").GetString());
+   
+    using (JsonDocument doc = JsonDocument.Parse(json))
+    {
+        return doc.RootElement.Clone();
+    }
+}
+```
+
+The preceding code expects a `JsonElement` that contains a `fileName` property. It opens the JSON file and creates a `JsonDocument`. The method assumes that the caller wants to work with the entire document, so it returns the `Clone` of the `RootElement`. 
+
+If you receive a `JsonElement` that isn't owned by a `JsonDocument` and return a sub-element, it's not necessary to return a `Clone` of the sub-element. For example:
```

Comment:
That isn't accurate. All `JsonElement` must by tied to a `JsonDocument` in some way. `Clone` effectively just makes it so that the `JsonDocument` that holds the `JsonElement` doesn't use the array pool and hence doesn't need to be disposed.

And you don't have to call `Clone` even in the case that `JsonDocument` is still disposable/uses the array pool **as long as you are returning a sub-element of the passed-in JsonElement**. The owner of the passed-in JsonElement continues to own the returned JsonElement. Essentially the "ownership" question is not relevant to the API in this scenario.

```suggestion
If you receive a `JsonElement` and are returning a sub-element, it's not necessary to return a `Clone` of the sub-element. The caller is responsible of keeping alive the `JsonDocument` that holds the passed-in `JsonElement`. For example:
```